modules/create_timepoint/jsx/createTimepointIndex.js

```diff
@@ -0,0 +1,390 @@
+    this.handleVisitLabel = this.handleVisitLabel.bind(this);
+    this.collectParams = this.collectParams.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.setForm = this.setForm.bind(this);
+  }
+
+  /**
+   * Executes after component mounts.
+   */
+  componentDidMount() {
+    this.collectParams();
+    this.fetchInitializerData()
+      .then(() => this.setState({isLoaded: true}));
+  }
+
+  /**
+   * Retrieve params from the browser URL and save it in state.
+   */
+  collectParams() {
+    const url = new URL(window.location.href);
+    const state = Object.assign({}, this.state);
+    state.url.params = {
+      candID: url.searchParams.get('candID'),
+      identifier: url.searchParams.get('identifier'),
+    };
+    state.data.dccid = state.url.params.candID;
+    this.setState(state);
+  }
+
+  /**
+   * Retrieve data from the provided URL and save it in state.
+   *
+   * @return {object}
+   */
+  fetchInitializerData() {
+    const state = Object.assign({}, this.state);
+    return fetch(
+      this.props.baseURL +
+      '/create_timepoint/Timepoint?candID=' + state.url.params.candID +
+      '&identifier=' + state.url.params.identifier,
+      {
+        credentials: 'same-origin',
+      }
+    ).then((response) => response.json())
+      .then((data) => {
+        const state = Object.assign({}, this.state);
+        // Populate the form conflicts.
+        if (data.conflict) {
+          state.messages = data.conflict;
+        }
+        // Populate the select options for subproject.
+        if (data.hasOwnProperty('subproject')) {
+          state.form.options.subproject = data.subproject;
+          state.form.value.subproject = null;
+          state.form.display.subproject = true;
+        }
+        // Populate the select options for psc.
+        if (data.psc) {
+          state.form.options.psc = data.psc;
+          state.form.value.psc = null;
+          state.form.display.psc = true;
+        }
+        // Populate the select options for project.
+        if (data.project) {
+          state.form.options.project = data.project;
+          state.form.value.project = null;
+          state.form.display.project = true;
+        }
+        // Populate the select options for visit.
+        if (data.visit) {
+          // Store the (complete) visit selection information.
+          state.storage.visit = data.visit;
+          // Handle visit selection.
+          this.handleVisitLabel();
+        }
+        // Populate the select options for languages.
+        if (data.languages) {
+          state.form.options.languages = data.languages;
+          state.form.value.languages = null;
+          state.form.display.languages = true;
+        }
+        this.setState(state);
+      }).catch((error) => {
+        const state = Object.assign({}, this.state);
+        state.messages = [{message: error}];
+        this.setState(state);
+      });
```

Comment:
I'm a little confused as to what's going on here. It seems like a lot of if statements for something we have control over sending from the backend. Is this just to load options? It seems like the `value` is always set to `null` and the `display` is always set to `true`, so would there be a way to write this more efficiently?

Reply:
@HenriRabalais the if statements are necessary I believe because I think the module changes depending on configurations.